packages/amplify-cli-core/API.md

```diff
@@ -1653,6 +1653,8 @@ export class PathManager {
     // (undocumented)
     getRootStackBuildDirPath: (projectPath: string) => string;
     // (undocumented)
+    getStackBuildCategorySourceDirPath: (projectPath: string, category: string, resourceName: string) => string;
```

Comment:
```suggestion
    getCategoryStackBuildDirPath: (projectPath: string, category: string, resourceName: string) => string;
```

Reply:
The path is `amplify/backend/awscloudformation/build/function/sourceID`, but `getCategoryStackBuildDirPath` seems like `amplify/backend/awscloudformation/build`

Reply:
`sourceID` in your example is `resourceName` no ?

Reply:
right, should be `resourceName`

Reply:
`amplify/backend/awscloudformation/build/function/resourceName` is presumably outcome of synthesis/"building resource". It's not "source directory" ?

Reply:
perhaps this should be `getResourceBuildDirPath` ?

Reply:
If we call `amplify/backend/awscloudformation/build/function/sourceName` as `getCategoryStackBuildDirPath`, what do we call `amplify/backend/awscloudformation/build`?

Reply:
@edwardfoyle what's best naming here?

Reply:
I'd go even simpler: `getCategoryBuildDir`

Reply:
> I'd go even simpler: `getCategoryBuildDir`

No, because
1. we've go `getRootStackBuildDirPath` for `amplify/backend/awscloudformation`
2. it'd be confused with `amplify/backend/function`

Reply:
In that case I think @sobolk suggestion of `getResourceBuildDirPath` makes the most sense

Reply:
No, `getCategoryStackBuildDirPath` could mean `amplify/backend/awscloudformation/build/function` , but we actually mean `amplify/backend/awscloudformation/build/function/<resourceName>`

Reply:
@0618 we're suggesting `getResourceBuildDirPath` not `getCategoryStackBuildDirPath`

Reply:
@sobolk would `getResourceBuildDirPath` be confused with `amplify/backend/function/<resourceName>`?

Reply:
`amplify/backend/function/<resourceName>` would be `getResourceDirPath` i.e. without `Build`

Reply:
Oh....I didn't noticed that it should be `resource` 😂

Reply:
updated d2827f36ecf3a711d1d0a88ef0703837268718a8

Reply:
doesn't look like the rename got pushed to the PR?